src/hotspot/share/memory/virtualspace.cpp

```diff
@@ -667,8 +669,8 @@ MemRegion ReservedHeapSpace::region() const {
 ReservedCodeSpace::ReservedCodeSpace(size_t r_size,
                                      size_t rs_align,
                                      size_t rs_page_size) : ReservedSpace() {
+  set_nmt_flag(mtCode);
```

Comment:
Why isn't this a part of the initialize call? This looks like a bug to me. `initialize` will call clear_members, which will undo this setting.

Reply:
`set_nmt_flag()`, `initialize` and `initialize_members` are chagned accorrding to the related comments.